adds NGFW support for google_network_security_tls_inspection_policy resource

maxi-cit opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->

Part of: https://github.com/hashicorp/terraform-provider-google/issues/15779

Adds support for NGFW, CA pools, Trust Configs and associated metadata

<!--
Please self-review your PR against the review checklist before creating it: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/

Completing the checklist will help speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.

If your PR is still work in progress, please create it in draft mode
-->

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/develop/run-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none

Unless you choose release-note:none, please add a release note.

See / for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
network-security: added `trust_config`, `min_tls_version`, `tls_feature_profile` and `custom_tls_features` fields to `google_network_security_tls_inspection_policy` resource
```

modular-magician commented:
Hello! I am a robot. It looks like you are a: Community Contributor ~Googler~ ~Core Contributor~. Tests will require approval to run.

@NickElliot, a repository maintainer, has been assigned to review your changes. If you have not received review feedback within 2 business days, please leave a comment on this PR asking them to take a look.

You can help make sure that review is quick by [doing a self-review](https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/) and by [running impacted tests locally](https://googlecloudplatform.github.io/magic-modules/get-started/run-provider-tests/).

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 20 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9864-old..auto-pr-9864) ( 2 files changed, 152 insertions(+))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 40 insertions(+))

## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_network_security_server_tls_policy` (7 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_network_security_server_tls_policy" "primary" {
  custom_tls_features = # value needed
  min_mtls_version    = # value needed
  tls_feature_profile = # value needed
  trust_config        = # value needed
}

```

modular-magician commented:
#### Tests analytics 
Total tests: `32` 
Passed tests `27` 
Skipped tests: `0` 
Affected tests: `5` 

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networksecurity</li></ul></blockquote></details> 

#### Action taken 
<details> <summary>Found 5 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyBasicExample|TestAccNetworkSecurityServerTlsPolicy_update|TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyServerCertExample|TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyAdvancedExample|TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyMtlsExample </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$ 
 `TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyBasicExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/build-log/recording_build/TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/recording/TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyBasicExample.log)]
`TestAccNetworkSecurityServerTlsPolicy_update`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/build-log/recording_build/TestAccNetworkSecurityServerTlsPolicy_update_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/recording/TestAccNetworkSecurityServerTlsPolicy_update.log)]
`TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyServerCertExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/build-log/recording_build/TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyServerCertExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/recording/TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyServerCertExample.log)]
`TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyAdvancedExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/build-log/recording_build/TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyAdvancedExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/recording/TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyAdvancedExample.log)]
`TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyMtlsExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/build-log/recording_build/TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyMtlsExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/recording/TestAccNetworkSecurityServerTlsPolicy_networkSecurityServerTlsPolicyMtlsExample.log)] 

$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$ 
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/51cd6a94-d585-4beb-a5bd-6fa19ba599b3/recording) for each test

NickElliot reviewed (commented):
Due to this fields not being supported by the API yet, the tests are erroring because it attempts to send that created default value. Do you have any documentation links that suggest when support may be enabled? I was unable to find anything linked in the referenced issue or through my own searches, despite seeing instructions for creating a policy with them [here](https://cloud.google.com/firewall/docs/setup-tls-inspection#configure-tls-inspection-policy).

maxi-cit commented:
Hello I am working on a few tests for these but I got delayed because I needed [this](https://github.com/GoogleCloudPlatform/magic-modules/pull/10122) to be merged first

shuyama1 commented:
@maxi-cit Sorry, there's an issue in our CI. Would you mind rebasing your PR to catch the latest fix? Thank you!

NickElliot commented:
/gcbrun

NickElliot commented:
/gcbrun

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 20 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9864-old..auto-pr-9864) ( 2 files changed, 153 insertions(+))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 40 insertions(+))

## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_network_security_tls_inspection_policy` (4 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_network_security_tls_inspection_policy" "primary" {
  custom_tls_features = # value needed
  min_tls_version     = # value needed
  tls_feature_profile = # value needed
  trust_config        = # value needed
}

```

modular-magician commented:
#### Tests analytics
Total tests: `37`
Passed tests: `33`
Skipped tests: `1`
Affected tests: `3`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networksecurity</li></ul></blockquote></details>

#### Action taken
<details> <summary>Found 3 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkSecurityGatewaySecurityPolicy_networkSecurityGatewaySecurityPolicyTlsInspectionBasicExample|TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyBasicExample|TestAccNetworkSecurityTlsInspectionPolicy_update </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccNetworkSecurityGatewaySecurityPolicy_networkSecurityGatewaySecurityPolicyTlsInspectionBasicExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/092c8abf-0a17-422a-99bf-417716f8b686/build-log/recording_build/TestAccNetworkSecurityGatewaySecurityPolicy_networkSecurityGatewaySecurityPolicyTlsInspectionBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/092c8abf-0a17-422a-99bf-417716f8b686/recording/TestAccNetworkSecurityGatewaySecurityPolicy_networkSecurityGatewaySecurityPolicyTlsInspectionBasicExample.log)]
`TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyBasicExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/092c8abf-0a17-422a-99bf-417716f8b686/build-log/recording_build/TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/092c8abf-0a17-422a-99bf-417716f8b686/recording/TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyBasicExample.log)]
`TestAccNetworkSecurityTlsInspectionPolicy_update`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/092c8abf-0a17-422a-99bf-417716f8b686/build-log/recording_build/TestAccNetworkSecurityTlsInspectionPolicy_update_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/092c8abf-0a17-422a-99bf-417716f8b686/recording/TestAccNetworkSecurityTlsInspectionPolicy_update.log)]


$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/092c8abf-0a17-422a-99bf-417716f8b686/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/092c8abf-0a17-422a-99bf-417716f8b686/recording) for each test

NickElliot reviewed (changes requested):
There is a permadiff in the tests because it seems the API does not return the fields when set to these default values. It be better to remove the "unspecified" values (and have no default set for these fields) as the fields are optional to begin with, and not computed, so the unspecified values will never be used by users.

maxi-cit commented:
Hello @NickElliot could you run the tests please?

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 53 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9864-old..auto-pr-9864) ( 3 files changed, 222 insertions(+))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 40 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `37`
Passed tests: `35`
Skipped tests: `1`
Affected tests: `1`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networksecurity</li></ul></blockquote></details>

#### Action taken
<details> <summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyBasicExample </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyBasicExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/689736d5-eebf-476c-a64f-e90670aecbc0/build-log/recording_build/TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/689736d5-eebf-476c-a64f-e90670aecbc0/recording/TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyBasicExample.log)]


$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/689736d5-eebf-476c-a64f-e90670aecbc0/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/689736d5-eebf-476c-a64f-e90670aecbc0/recording) for each test

NickElliot reviewed (changes requested):
Can you move the basic test additions you made to a new or different test? The basic test is meant to me a minimal example that does not have more fields than what is need to make the resource period.

maxi-cit commented:
Hello @NickElliot I updated the PR based on your suggestions. So, I added a separate example to test these new fields. However The test should be failing because of a projectId/Number issue regarding the trust_config field.
```
|       ~ trust_config          = "projects/29670186149/locations/us-central1/trustConfigs/tf-test-my-trust-configppxkl6y7pl" -> "projects/prj-cfio-mp-test008-svc-ext/locations/us-central1/trustConfigs/tf-test-my-trust-configppxkl6y7pl"
```
I already reported and they told me they are fixing it soon. Once its fixed tests should be passing

maxi-cit commented:
Hello @NickElliot, could you run the tests please?

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 160 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9864-old..auto-pr-9864) ( 3 files changed, 456 insertions(+))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 40 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `38`
Passed tests: `36`
Skipped tests: `1`
Affected tests: `1`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networksecurity</li></ul></blockquote></details>

#### Action taken
<details> <summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyCustomExample </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyCustomExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/d552ae35-925b-428b-9bed-d29ea315929a/build-log/recording_build/TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyCustomExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/d552ae35-925b-428b-9bed-d29ea315929a/recording/TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyCustomExample.log)]


$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/d552ae35-925b-428b-9bed-d29ea315929a/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/d552ae35-925b-428b-9bed-d29ea315929a/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 164 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9864-old..auto-pr-9864) ( 3 files changed, 464 insertions(+))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 40 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `38`
Passed tests: `36`
Skipped tests: `1`
Affected tests: `1`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networksecurity</li></ul></blockquote></details>

#### Action taken
<details> <summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyCustomExample </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$
`TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyCustomExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/26a8f393-5c1b-4599-b3f0-c303307c7950/recording/TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyCustomExample.log)]


$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$

---
$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/26a8f393-5c1b-4599-b3f0-c303307c7950/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/26a8f393-5c1b-4599-b3f0-c303307c7950/recording) for each test

LucaPrete commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,136 @@
+  service = "networksecurity.googleapis.com"
+}
+
+resource "google_privateca_ca_pool_iam_member" "tls_inspection_permission" {
```
should we call this `default` as all other resources?

maxi-cit replied:
which resource are u referring to?

LucaPrete replied:
line 74. I would replace the resource name `tls_inspection_permission` with `default` to be consistent with other resource names you declared here.

LucaPrete commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,136 @@
+
+resource "google_project_service_identity" "ns_sa" {
+  provider = google-beta
+
```
I'd remove to be consistent all these blank lines between the provider meta-argument and all other arguments

maxi-cit replied:
done

LucaPrete commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,136 @@
+    }
+  }
+}
+
```
we should leave only one line between resource declarations

maxi-cit replied:
done

LucaPrete commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,136 @@
+  deletion_protection = false
+  skip_grace_period = true
+  ignore_active_certificates_on_deletion = true
+  config {
```
we should leave one line between single line arguments and blocks

maxi-cit replied:
done

LucaPrete commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,136 @@
+
+  labels = {
+    foo = "bar"
+  }
```
unless we're not testing the `google_certificate_manager_trust_config ` resource I'd keep it as minimal as possible. For example, I wouldn't put `labels = {...}`

LucaPrete commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,136 @@
+    "TLS_RSA_WITH_AES_256_GCM_SHA384",
+  ]
+
+  depends_on = [
```
All these shouldn't be needed anymore if you use the implicit dependencies recommended above.

maxi-cit replied:
I put those just in case of the replay test

LucaPrete replied:
@maxi-cit not sure what you mean. I'll let @NickElliot elaborate on this.
For sure it's not something I see super often in our code. I'd avoid it as much as possible.

maxi-cit replied:
@LucaPrete there are times where you need to be explicit about the implicit specially when you create similar resources. And this is due some REPLAYING check where resources should be deleted the opposite way they were created. At least, it is what I understand.

LucaPrete commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,136 @@
+resource "google_privateca_ca_pool_iam_member" "tls_inspection_permission" {
+  provider = google-beta
+
+  ca_pool = google_privateca_ca_pool.default.id
```
Try to make this `google_privateca_certificate_authority.default.pool`.
Not sure if the name will work. Alternatively, you can try to pass the id instead of the pool name in `google_privateca_certificate_authority`.

If you make this work, all explicit dependencies below (depends_on) will go away.

maxi-cit replied:
google_privateca_certificate_authority does not work with id

LucaPrete reviewed (changes requested):
I think we need a few changes and I'd also recommend to add a handwritten test for the update.

maxi-cit commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,136 @@
+  exclude_public_ca_set = false
+
+  min_tls_version     = "TLS_1_0"
+  trust_config        = "projects/${data.google_project.project.number}/locations/${google_certificate_manager_trust_config.default.location}/trustConfigs/${google_certificate_manager_trust_config.default.name}"
```
already tried it does not work due to the project ID/number issue. they told me they will be fixing this but havent got any news so far

NickElliot replied:
This should be implemented as a diff_suppress_func (see: https://googlecloudplatform.github.io/magic-modules/develop/permadiff/#normalized-value) that automatically parses an equivalency between using number and using name. This would allow the ID reference to be used as customers will expect.

NickElliot reviewed (changes requested):
If the API won't return the value to allow users to use a basic ID reference this will need to be implemented with a diff_suppress_func, https://googlecloudplatform.github.io/magic-modules/develop/permadiff/#normalized-value

maxi-cit commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,136 @@
+data "google_project" "project" {
```
removed

maxi-cit commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,136 @@
+
+resource "google_privateca_ca_pool" "default" {
+  provider = google-beta
+  name      = "<%= ctx[:vars]['privateca_ca_pool_name'] %>"
```
done

maxi-cit commented:
Hello @NickElliot, I added most of the suggestions. Could you give it a new round of review please?

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9864-old..auto-pr-9864) ( 2 files changed, 295 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9864-old..auto-pr-9864) ( 3 files changed, 457 insertions(+))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 40 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `38`
Passed tests: `36`
Skipped tests: `1`
Affected tests: `1`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networksecurity</li></ul></blockquote></details>


#### Action taken
<details> <summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyCustomExample </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$
`TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyCustomExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/327dd1f7-9ca3-4d57-b0d2-5fd6de2daf14/recording/TestAccNetworkSecurityTlsInspectionPolicy_networkSecurityTlsInspectionPolicyCustomExample.log)]


$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$

---
$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/327dd1f7-9ca3-4d57-b0d2-5fd6de2daf14/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/327dd1f7-9ca3-4d57-b0d2-5fd6de2daf14/recording) for each test

NickElliot reviewed (changes requested):
a couple of these explicit dependencies are unneeded

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9864-old..auto-pr-9864) ( 2 files changed, 295 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9864-old..auto-pr-9864) ( 3 files changed, 457 insertions(+))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 40 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `38`
Passed tests: `37`
Skipped tests: `1`
Affected tests: `0`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networksecurity</li></ul></blockquote></details>$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/cd1905a2-ddaa-497f-ad30-3079245f8303/build-log/replaying_test.log)

maxi-cit commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,132 @@
+    google_privateca_ca_pool.default,
+    google_privateca_certificate_authority.default,
+    google_privateca_ca_pool_iam_member.default,
+    google_certificate_manager_trust_config.default
```
removed

maxi-cit commented on mmv1/templates/terraform/examples/network_security_tls_inspection_policy_custom.tf.erb:
```diff
@@ -0,0 +1,132 @@
+  ]
+
+  depends_on = [
+    google_privateca_ca_pool.default,
```
removed

maxi-cit commented:
@NickElliot I removed the dependencies could you run the tests, please?

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-9864-old..auto-pr-9864) ( 2 files changed, 293 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9864-old..auto-pr-9864) ( 3 files changed, 453 insertions(+))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-9864-old..auto-pr-9864) ( 1 file changed, 40 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `38`
Passed tests: `37`
Skipped tests: `1`
Affected tests: `0`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networksecurity</li></ul></blockquote></details>

$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-9864/artifacts/dddafaa3-92a0-4cae-bef2-e0c66b978b0c/build-log/replaying_test.log)

NickElliot reviewed (approved):
LGTM!
